org.jacoco.agent.rt.test/src/org/jacoco/agent/rt/internal/output/TcpClientOutputTest.java

```diff
@@ -81,17 +82,25 @@ public void testRemoteClose() throws Exception {
 	}
 
 	@Test
-	public void testInvalidCommand() throws Exception {
+	public void should_log_exception_when_invalid_command_is_sent()
+			throws Exception {
 		remoteWriter.visitSessionInfo(new SessionInfo("info", 1, 2));
-		while (remoteReader.read()) {
+
+		try {
+			while (remoteReader.read()) {
+			}
+		} catch (EOFException e) {
+			// Might be thrown as the socket is directly closed by the client
```

Comment:
@Godin Now thst you're asking I realize that I haven't fully understood the reason neither :flushed: 

I'll investigate further and provide a updated commit. At least I can reproduce it locally from time to time.